src/test/e2e/api/admin/state.e2e.test.ts

```diff
@@ -464,3 +466,33 @@ test(`should not show environment on feature toggle, when environment is disable
     expect(result[1].name).toBe('production');
     expect(result[1].enabled).toBeFalsy();
 });
+
+test(`should handle v3 export with variants in features`, async () => {
+    app.services.stateService = new StateService(db.stores, {
+        getLogger,
+        flagResolver: {
+            isEnabled: () => false,
+            getAll: () => ({}),
+        },
+    });
+
+    await app.request
+        .post('/api/admin/state/import?drop=true')
+        .attach('file', 'src/test/examples/variantsexport_v3.json')
+        .expect(202);
+
+    const exported = await app.services.stateService.export({});
+    let exportedFeatures = exported.features
+        .map((f) => {
+            delete f.createdAt;
+            return f;
+        })
+        .sort();
+    let importedFeatures = variantsv3.features
+        .map((f) => {
+            delete f.createdAt;
+            return f;
+        })
+        .sort();
+    expect(exportedFeatures).toStrictEqual(importedFeatures);
+});
```

Comment:
we probably should test all variants of the export with the old format, to make sure we handle things being turned on/off?

Reply:
Then again, we have really only changed the logic for feature and featureEnvironment, so maybe I'm being overly defensive